File: killrvideo-service-comments/src/main/java/com/killrvideo/service/comment/dao/CommentRowMapper.java
package com.killrvideo.service.comment.dao;

import com.datastax.oss.driver.api.core.cql.Row;
import com.killrvideo.service.comment.dto.Comment;
import org.springframework.stereotype.Component;

import static com.killrvideo.service.comment.dto.Comment.*;

@Component
public class CommentRowMapper {
    public Comment map(Row row) {
        Comment c = new Comment();
        c.setComment(row.getString(COLUMN_COMMENT));
        c.setUserid(row.getUuid(COLUMN_USERID));
        c.setCommentid(row.getUuid(COLUMN_COMMENTID));
        c.setVideoid(row.getUuid(COLUMN_VIDEOID));
        c.setDateOfComment(row.getInstant("comment_timestamp"));
        return c;
    }
}

File: killrvideo-service-comments/src/main/kotlin/com/killrvideo/service/comment/dao/CommentRowMapper.kt
package com.killrvideo.service.comment.dao

import com.datastax.oss.driver.api.core.cql.Row
import com.killrvideo.service.comment.dto.Comment
import org.springframework.stereotype.Component

@Component
class CommentRowMapper {
    fun map(row: Row): Comment =
        Comment(
            comment = row.getString(Comment.COLUMN_COMMENT),
            userid = row.getUuid(Comment.COLUMN_USERID),
            commentid = row.getUuid(Comment.COLUMN_COMMENTID),
            videoid = row.getUuid(Comment.COLUMN_VIDEOID),
            dateOfComment = row.getInstant(Comment.COLUMN_COMMENT_TIMESTAMP)
        )
}
